impl/src/main/java/org/ehcache/internal/classes/ClassInstanceProvider.java

```diff
@@ -68,15 +68,27 @@ protected T newInstance(String alias, ServiceConfiguration<?> serviceConfigurati
     return newInstance(alias, config);
   }
 
-  private T newInstance(String alias, ClassInstanceConfiguration<T> config) {
+  protected T newInstance(String alias, Object ... args) {
+    return newInstance(alias, null, args);
+  }
+
+  private T newInstance(String alias, ClassInstanceConfiguration<T> config, Object... args) {
     if (config == null) {
       config = getPreconfigured(alias);
       if (config == null) {
         return null;
       }
     }
+    final Object[] constructorArgs = args.length > 0 ? args : config.getArguments();
```

Comment:
The contract here is enforced by the two protected method calling in, i.e. either
 - the config passed in was null, so we _ignore_ possible preconfigured config args
 - the vararg was empty, so we use the config (preconfigured or not)

Not sure how happy I am with this contract